### ISSUE TITLE
Long Rmd chunk output causes RStudio to hang when opening file / session

### ISSUE THREAD

**Issue by dfalty** [rstudio/rstudio#5518]
### System details

    RStudio Edition : Desktop and Server
    RStudio Version : 1.2.1578-4 & 1.3.482-1

### Steps to reproduce the problem
1. Create a rmd file with the following chunk
````markdown
```{r}
for (i in 1:50000) {
  print(i)
}
```
````
2. Run the chunk
3. Close the file
4. Open the file
5. Observe very long hang until RStudio is responsive

### Describe the problem in detail
Long Rmd chunk output causes RStudio to hang when opening file / session. Happens on desktop and server but server seems worse because you continually get that the browser is unresponsive messages. I'm hoping there is something we can do to make this experience better.
This was found in a rstudio.cloud project. When the user opened the rstudio.cloud project, the RStudio session started with the file already opened and froze the session immediately.

**Comment by dfalty** [comment]
Changing to the 1.2-patch for discussion since there was a large class that was affected by this on rstudio.cloud.

**Comment by jmcphers** [comment]
Backported to 1.2-patch in https://github.com/rstudio/rstudio/commit/bcdaf575055457df4dc4bc55f524dc10c0f85f41

**Comment by dfalty** [comment]
Verified fixed 1.2.5016-1

**Comment by rguliev** [comment]
Hello,

I am facing the same problem. Is there any available workaround?
My assumption is that the report contains too many `ggplot`s.
```
R version: 4.1.0
Rstudio version: 1.4.1717
Rmd file size: 10.2Kb
built HTML report size: 1.5Mb
```
